plugins/hypervisors/xenserver/src/com/cloud/hypervisor/xenserver/resource/CitrixResourceBase.java

```diff
@@ -2590,7 +2590,9 @@ public SR getISOSRbyVmName(final Connection conn, final String vmName) {
     public VDI getIsoVDIByURL(final Connection conn, final String vmName, final String isoURL) {
         SR isoSR = null;
         String mountpoint = null;
-        if (isoURL.startsWith("xs-tools")) {
+        // XenServer 7.0+ => guest-tools.iso
+        // XenServer [other] => xs-tools.iso
+        if (isoURL.startsWith("xs-tools") || isoURL.startsWith("guest-tools")) {
```

Comment:
Is it possible for `isoURL` to be null? If not, it is ok the way it is. Otherwise, it might be a good idea to use StringUtils.startsWith

Reply:
+1 I doubt it will be null, but checking for null almost always is good. I'll add it.